woocommerce/woocommerce
# Add category field dropdown field

### All Submissions:






### Changes proposed in this Pull Request:

Adds a new Category dropdown field that includes a nested tree for the categories.

**Changes done as part of this**
- (Part of https://github.com/woocommerce/woocommerce/pull/34620 now ) Update experimental select control component to return more callbacks from downshift and allow the option to keep the dropdown open when an item is selected.
- (part of https://github.com/woocommerce/woocommerce/pull/34613 now ) Added `getItemsTotalCount` to the crud data store, allowing us to use it for categories - `getProductCategoriesTotalCount`
- (will also be part of a separate PR )  Add a new Category dropdown to the MVP product page (this includes several new components).

Addresses: https://github.com/woocommerce/woocommerce/issues/34029

https://user-images.githubusercontent.com/2240960/188210707-414a57a4-1f26-4b6f-a69d-b8206e0cfc16.mp4

### How to test the changes in this Pull Request:

Should match these designs: 5sAIeTRd9Yp7nSCT33BAWz-fi-5175%3A105193

1. Load this branch and enable the `new-product-management-experience` feature flag by using the WCA Test Helper (Tools > WCA Test Helper > Features) and refresh the page.
2. Create some categories under **Product > Categories**
3. Click the **Products > Add New (MVP)** button and notice the new category field dropdown.
4. Start searching in the dropdown field and make sure your created categories show up
5. See if categories are correctly sorted by popularity first and then alphabetical order (popularity is by how many products its being used).
6. Select one of the child categories, it should auto select the parents but allow you to un-select the parents
7. You should also be able to use your keyboard by using your up and down arrows, currently I have the selected item highlighted by setting it to bold and have closed parents automatically open if they have children.

### Other information:




### FOR PR REVIEWER ONLY:
